x/concentrated-liquidity/math/math_test.go

```diff
@@ -156,6 +162,37 @@ func (suite *ConcentratedMathTestSuite) TestGetNextSqrtPriceFromAmount1RoundingD
 	}
 }
 
+// TestGetNextSqrtPriceFromAmount1RoundingDownBigDec tests that getNextSqrtPriceFromAmount1RoundingDown
+// utilizes the current squareRootPrice, liquidity of denom1, and amount of denom1 that still needs
+// to be swapped in order to determine the next squareRootPrice
+// sqrtPriceNext = sqrtPriceCurrent + (amount1Remaining / liquidity1)
+func (suite *ConcentratedMathTestSuite) TestGetNextSqrtPriceFromAmount1RoundingDownBigDec() {
```

Comment:
Note: both tests added in this file are 1:1 copies with final results recalculated in Python